src/doc/unstable-book/src/library-features/asm.md

```diff
@@ -345,6 +345,29 @@ The `h` modifier will emit the register name for the high byte of that register
 
 If you use a smaller data type (e.g. `u16`) with an operand and forget the use template modifiers, the compiler will emit a warning and suggest the correct modifier to use.
 
+## Memory address operands
+
+Sometimes assembly instructions require operands passed via memory addresses/memory locations.
+You have to manually use the memory address syntax specified by the respectively architectures.
+For example, in x86/x86_64 and intel assembly syntax, you could wrap inputs/outputs in `[]`
+to indicate they are memory operands.
+
+Old way:
+```rust,allow_fail
+# #![feature(llvm_asm)]
+# fn load_fpu_control_word(control: u16) {
+unsafe { llvm_asm!("fldcw $0" :: "m" (control) :: "volatile") }
+# }
+```
+
+New way:
+```rust,allow_fail
+# #![feature(asm)]
+# fn load_fpu_control_word(control: u16) {
+unsafe { asm!("fldcw [{}]", in(reg) &control, options(nostack)) }
```

Comment:
What happens if one does not use `&`? Will it error or it will just UB?

Reply:
Not error or UB, just incorrect.

Reply:
Huh? So it still works?

Reply:
No it can't possibly work. It will crash at runtime since the value of `control` is not a valid memory address.

Reply:
But it won't be compile time error but just a panic at runtime?

Reply:
In the most and best case it is a crash (seg fault).